### PR TITLE
More additions to typed api

### PR DESCRIPTION
**Pull request by Jimbo4350**

### CODE REVIEW

File: cardano-api/src/Cardano/Api/Typed.hs [hunk 1]
```diff
@@ -91,7 +91,7 @@ module Cardano.Api.Typed (
     makeByronTransaction,
     makeShelleyTransaction,
     SlotNo,
-    TxOptional(..),
+    TxExtraContent(..),
```
Comment (dcoutts) [inline cardano-api/src/Cardano/Api/Typed.hs:94]: Do you like this name? I'm unsure.

I know I was the one who changed it :-)

Comment (Jimbo4350) [reply cardano-api/src/Cardano/Api/Typed.hs:94]: The only other name I can think of is `TxAdditional`. Both work for me.

Comment (dcoutts) [reply cardano-api/src/Cardano/Api/Typed.hs:94]: It's not yet final, it's fine. We can still decide what colour we like best.

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**